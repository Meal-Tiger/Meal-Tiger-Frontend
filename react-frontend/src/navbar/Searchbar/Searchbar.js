import 'material-symbols';
import styles from './Searchbar.module.css';

function Searchbar(){
    return (
        <div>
            <form className={styles.form}>
                <input className={styles.input} type="text" placeholder="Search.."></input>
                <button className={styles.button} type="submit"><span className="material-symbols-outlined">search</span></button>
            </form>
        </div>
        
    );
}

export default Searchbar;