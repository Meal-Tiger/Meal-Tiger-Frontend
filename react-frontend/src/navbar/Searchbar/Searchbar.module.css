.form{
    display: flex;
    align-items: center;
}

.input {
    padding: 5px 0;
}

.button{
    height: 100%;
}