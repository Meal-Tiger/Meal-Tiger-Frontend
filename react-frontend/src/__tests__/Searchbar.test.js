import { render } from '@testing-library/react';
import Searchbar from "../navbar/Searchbar/Searchbar";

jest.mock('react-router-dom', () => {
    return {
        ...jest.requireActual('react-router-dom'),
        useNavigate: jest.fn()
    }
});

test('Searchbar renders correctly', () => {
    const { container } = render(<Searchbar />);
    expect(container).toMatchSnapshot();
});
