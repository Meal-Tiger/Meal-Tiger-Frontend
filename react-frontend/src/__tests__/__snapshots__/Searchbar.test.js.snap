// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searchbar renders correctly 1`] = `
<div>
  <form
    class="form"
  >
    <input
      placeholder="Search.."
      type="text"
    />
    <button
      class="btn btn-icon"
      type="submit"
    >
      <span
        class="material-symbols-outlined"
      >
        search
      </span>
    </button>
  </form>
</div>
`;
